src/utils/component-swap-data.ts

```diff
@@ -238,10 +245,11 @@ function getIssuanceContract(
 
 async function getUnderlyingErc20(
   token: string,
-  chainId: number
+  chainId: number,
+  rpcUrl: string
 ): Promise<WrappedToken> {
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const publicClient = createClient(chainId)!
+  const publicClient = createClientWithUrl(chainId, rpcUrl)!
```

Comment:
The issue was the `createClient` would try to use a internal env var which is not available when use from clients like the API. So now we're passing the RPC URL.